File: CS/DXSampleSplashScreenService/ViewModel/MainViewModel.cs
using DevExpress.Mvvm;
using DevExpress.Mvvm.POCO;
using DevExpress.Mvvm.DataAnnotations;
using System;
using System.Threading;

namespace DXSampleSplashScreenService.ViewModel {
    [POCOViewModel]
    public class MainViewModel {
        public virtual int Delay { get; set; }
        protected ISplashScreenService SplashScreenService { get { return this.GetService<ISplashScreenService>(); } }
        
        public MainViewModel() {
            Delay = 10;
        }

        public void Calculate() {
            SplashScreenService.ShowSplashScreen();
            SplashScreenService.SetSplashScreenState("Calculating...");
            Thread.Sleep(TimeSpan.FromSeconds(Delay));
            SplashScreenService.SetSplashScreenState("Finished.");
            Thread.Sleep(TimeSpan.FromSeconds(2));
            SplashScreenService.HideSplashScreen();
        }
    }
}

File: VB/DXSampleSplashScreenService/ViewModel/MainViewModel.vb
Imports DevExpress.Mvvm
Imports DevExpress.Mvvm.POCO
Imports DevExpress.Mvvm.DataAnnotations
Imports System
Imports System.Threading

Namespace DXSampleSplashScreenService.ViewModel

    <POCOViewModel>
    Public Class MainViewModel

        Public Overridable Property Delay As Integer

        Protected ReadOnly Property SplashScreenService As ISplashScreenService
            Get
                Return GetService(Of ISplashScreenService)()
            End Get
        End Property

        Public Sub New()
            Delay = 10
        End Sub

        Public Sub Calculate()
            SplashScreenService.ShowSplashScreen()
            SplashScreenService.SetSplashScreenState("Calculating...")
            Thread.Sleep(TimeSpan.FromSeconds(Delay))
            SplashScreenService.SetSplashScreenState("Finished.")
            Thread.Sleep(TimeSpan.FromSeconds(2))
            SplashScreenService.HideSplashScreen()
        End Sub
    End Class
End Namespace
